### ISSUE TITLE
Feature Request: Retry segment download on timeout.

### ISSUE THREAD

**Issue by schumi4** [gabe565/CastSponsorSkip#15]
When a video is already running on boot (like when having an RPi powered by the TV itself) the segment download fails:

```
Aug 28 11:29:41 DietPi systemd[1]: Started castsponsorskip.service - A Go program that skips sponsored YouTube content on all local Google Cast devices.
Aug 28 11:29:43 DietPi castsponsorskip[429]: 2023/08/28 11:29:43 INFO Searching for devices...
Aug 28 11:29:43 DietPi castsponsorskip[429]: 2023/08/28 11:29:43 INFO Connected to cast device. device="Chromeo ULTRA"
Aug 28 11:29:44 DietPi castsponsorskip[429]: 2023/08/28 11:29:44 INFO Detected video stream. device="Chromeo ULTRA" video_id=[redacted]
Aug 28 11:29:54 DietPi castsponsorskip[429]: 2023/08/28 11:29:54 ERROR Failed to query segments. Retrying... device="Chromeo ULTRA" error="Get \"https://sponsor.ajay.app/api/skipSegments/[redacted]?category=sponsor&category=selfpromo&category=interaction\": net/http: TLS handshake timeout"
```
It says retrying, but does not seem to do that. Might be specific to my setup, but I can imagine segment download could also fail for numerous other temporary reasons. Maybe add limited retries on fail?

Thank you for considering!

**Comment by gabe565** [comment]
I don't think this happened due to the video already running on boot, but you're right that it should retry like the log says. It originally did but after a couple of optimization stages, it seems like it only tries once.

**Comment by gabe565** [comment]
I remember the change that caused this! It would originally retry on every update loop, but that caused streams with no segments to be retried every second. I just need to make it so that it retries if the segment download endpoint fails, but doesn't retry if it returns a 404!

**Comment by gabe565** [comment]
Fixed in [v0.3.3](https://github.com/gabe565/CastSponsorSkip/releases/tag/v0.3.3)
